usso/openid.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package usso
+
+import (
+	"html/template"
+	"net/http"
+	"strings"
+
+	"github.com/CanonicalLtd/serial-vault/datastore"
+	"github.com/juju/usso"
+	"github.com/juju/usso/openid"
+)
+
+var (
+	teams    = "ce-web-logs,canonical"
+	required = "email,fullname,nickname"
+	optional = ""
+)
+
+var client = openid.NewClient(usso.ProductionUbuntuSSOServer, &datastore.OpenidNonceStore, nil)
```

Comment:
I guess that client already knows the endpoint to be redirected (Location header) for open id authentication.
In that case, what happens if that url changes? (a preproduction openid environment, etc..). Or is that configurable somewhere that I've missed?
(btw, in Multitenancy doc maybe it would be nice having a schema of data exchange in openid login phase)